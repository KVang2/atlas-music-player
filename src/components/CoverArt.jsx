import React from 'react';
import image from '../assets/placeholder.svg';

function CoverArt() {
    return (
        <div>
            <img src={image} alt="PlaceHolder" />
        </div>
    );
}

export default CoverArt;