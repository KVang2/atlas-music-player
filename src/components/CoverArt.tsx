interface CoverArtProps {
  cover: string;
}

export default function CoverArt({ cover }: CoverArtProps) {
  return (
    <div className="flex justify-center">
      <img 
        src={cover}
        alt="Cover Art"
        className="w-[400px] h-[400px] object-cover rounded-md border border-gray-300" 
      />
    </div>
  );
}
