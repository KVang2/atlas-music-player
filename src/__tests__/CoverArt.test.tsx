import { render, screen } from "@testing-library/react";
import CoverArt from "../components/CoverArt";
import { describe, test, expect } from "vitest";

describe("CoverArt", () => {
    test("renders cover image correctly", () => {
        const testCover = "https://exmple.com/test-cover.jpg";
        const testSongId = "12345";

        render(<CoverArt cover={testCover} songId={testSongId} />);

        const image = screen.getByRole("img");
        expect(image).toBeInTheDocument();

        expect(image).toHaveAttribute("src", testCover);
        expect(image).toHaveAttribute("alt", "Cover Art");
    });

    test("matches snapshot", () => {
        const { container } = render(<CoverArt cover="https://example.com/test-cover.jpg" songId="12345" />);
        expect(container).toMatchSnapshot();
      });
    });