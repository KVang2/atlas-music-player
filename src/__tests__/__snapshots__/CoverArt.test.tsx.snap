// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt > matches snapshot 1`] = `
<div>
  <div
    class="relative flex justify-center"
  >
    <img
      alt="Cover Art"
      class="w-[450px] h-[450px] object-cover rounded-md border border-gray-300"
      src="https://example.com/test-cover.jpg"
    />
  </div>
</div>
`;
